pkg/cluster/serviceprincipal.go

```diff
@@ -48,27 +44,21 @@ func (m *manager) clusterSPObjectID(ctx context.Context) error {
 	// wait.PollImmediateUntil. Doing this will not propagate the latest error
 	// to the user in case when wait exceeds the timeout
 	_ = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
-		var res azgraphrbac.ServicePrincipalObjectResult
-		res, err = m.spApplications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
+		clusterSPObjectID, err = utilgraph.GetServicePrincipalIDByAppID(ctx, m.spGraphClient, spp.ClientID)
 		if err != nil {
-			// TODO - with migration to MSGraph, this error is different now.
-			// it's fixed with a refreshing action, but we should fix it here too
-			if strings.Contains(err.Error(), "Authorization_IdentityNotFound") {
```

Comment:
do you know what this error message is now?  

We're "artifically" waiting with the RefreshingAction change I made in #2944, but ideally we'd wait until the Application was fully propagated through ARM cache.

Reply:
No I don't.  It will be one of the error codes [listed here](https://learn.microsoft.com/en-us/graph/errors#code-property), but until we see this in the wild I can only guess.